### PR TITLE
compose: Add types to useReducedMotion and useMediaQuery

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Adds types to `useMediaQuery` and `useReducedMotion`. Only one runtime change was necessary to cast the return type of `useMediaQuery` to a boolean.

Also required adding a new variable to `gutenberg-env` and adding the reference to it to the `types` property of `compose`'s `tsconfig.json`.

Part of #18838

## How has this been tested?
Type checks pass and unit tests pass.

## Types of changes
New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/compose/src/hooks/use-media-query/index.js [hunk 1]
```diff
@@ -33,5 +33,5 @@ export default function useMediaQuery( query ) {
 		};
 	}, [ query ] );
 
-	return query && match;
+	return !! query && match;
```
Comment (tyxla) [inline packages/compose/src/hooks/use-media-query/index.js:36]: This seems to be breaking some existing tests.

Comment (sarayourfriend) [reply packages/compose/src/hooks/use-media-query/index.js:36]: Hmm, you're right. I wonder if we should update the test or keep the existing behavior and use a wider return type 🤔

Comment (sarayourfriend) [reply packages/compose/src/hooks/use-media-query/index.js:36]: I've gone ahead and updated the test.

File: packages/compose/src/hooks/use-media-query/test/index.js [hunk 2]
```diff
@@ -138,7 +138,8 @@ describe( 'useMediaQuery', () => {
 		await act( async () => {
 			root = create( <TestComponent /> );
 		} );
-		expect( root.toJSON() ).toBe( 'useMediaQuery: undefined' );
+		// query will be case to a boolean to simplify the return type.
```
Comment (ciampo) [inline packages/compose/src/hooks/use-media-query/test/index.js:141]: Nit: typo

```suggestion
		// query will be cast to a boolean to simplify the return type.
```

**Review by tyxla — APPROVED** [review]
Looks good to me 👍  I wonder if any of the failing tests could be related 🤔 I doubt they are TBH.

**Review by ntsekouras — APPROVED** [review]

**Pull request merged**